### PR TITLE
Add Last Value aggregation.

### PR DESCRIPTION
**Pull request by isturdy**

### CODE REVIEW

File: opencensus/exporters/stats/prometheus/internal/prometheus_utils.cc [hunk 1]
```diff
@@ -48,21 +48,39 @@ io::prometheus::client::MetricType MetricType(
...
   }
 }
 
 void SetValue(double value, io::prometheus::client::MetricType type,
               io::prometheus::client::Metric* metric) {
-  metric->mutable_untyped()->set_value(value);
+  if (type == io::prometheus::client::MetricType::UNTYPED) {
+    metric->mutable_untyped()->set_value(value);
+  } else {
+    metric->mutable_gauge()->set_value(value);
```
Comment (g-easy) [inline opencensus/exporters/stats/prometheus/internal/prometheus_utils.cc]: Please assert type is GAUGE.

File: opencensus/exporters/stats/prometheus/internal/prometheus_utils_test.cc [hunk 2]
```diff
@@ -157,6 +157,78 @@ TEST(SetMetricFamilyTest, SumInt) {
   )");
 }
 
+TEST(SetMetricFamilyTest, LastValueDouble) {
+  const auto measure = opencensus::stats::MeasureDouble::Register(
+      "measure_last_value_int", "", "units");
```
Comment (g-easy) [inline opencensus/exporters/stats/prometheus/internal/prometheus_utils_test.cc]: _int -> _double

File: opencensus/exporters/stats/prometheus/internal/prometheus_utils_test.cc [hunk 3]
```diff
@@ -157,6 +157,78 @@ TEST(SetMetricFamilyTest, SumInt) {
...
+        label { name: "foo" value: "v1" }
+        label { name: "bar" value: "v2" }
+        gauge { value: 2.0 }
+      }
+  )");
+}
+
+TEST(SetMetricFamilyTest, LastValueInt64) {
+  const auto measure = opencensus::stats::MeasureInt64::Register(
+      "measure_last_value_double", "", "units");
```
Comment (g-easy) [inline opencensus/exporters/stats/prometheus/internal/prometheus_utils_test.cc]: _double -> _int

:)

File: opencensus/stats/internal/view_data_impl.cc [hunk 4]
```diff
@@ -172,13 +178,30 @@ void ViewDataImpl::Merge(const std::vector<std::string>& tag_values,
   end_time_ = std::max(end_time_, now);
   switch (type_) {
     case Type::kDouble: {
-      double_data_[tag_values] += data.sum();
+      if (aggregation_.type() == Aggregation::Type::kSum) {
+        double_data_[tag_values] += data.sum();
+      } else {
+        double_data_[tag_values] = data.last_value();
```
Comment (g-easy) [inline opencensus/stats/internal/view_data_impl.cc]: assert type is kLastValue

**Comment by g-easy** [comment]
I don't understand why travis failed, `//opencensus/exporters/stats/stackdriver:stackdriver_utils_test` passed for me locally.

**Comment by g-easy** [comment]
It passes with gcc but fails with clang.

```
[ RUN      ] StackdriverUtilsTest.SetMetricDescriptorMetricKind

Program received signal SIGILL, Illegal instruction.
0x00007fb7ea4fd4b3 in opencensus::exporters::stats::(anonymous namespace)::GetValueType(opencensus::stats::ViewDescriptor const&) ()
```

Awesome.

File: opencensus/exporters/stats/prometheus/internal/prometheus_utils_test.cc (hunk 3 above)
Comment (isturdy) [reply opencensus/exporters/stats/prometheus/internal/prometheus_utils_test.cc]: Oops--done.

File: opencensus/stats/internal/view_data_impl.cc (hunk 4 above)
Comment (isturdy) [reply opencensus/stats/internal/view_data_impl.cc]: Done.

**Comment by isturdy** [comment]
And I cannot build locally with clang because of https://github.com/bazelbuild/bazel/issues/3977 re-arising. :(

**Comment by g-easy** [comment]
opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc:85 - we fall out of the outermost switch() without returning anything.

clang guards this by inserting a `ud2` instruction.

There's even a warning about it but I guess it got lost in the noise previously. :(

```
opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc:72:36: warning: enumeration value 'kLastValue' not handled in switch [-Wswitch]
  switch (descriptor.aggregation().type()) {
```

**Comment by g-easy** [comment]
BTW this works for me on my workstation:

```
bazel clean --expunge
env CC=clang bazel build -c dbg //opencensus/exporters/stats/stackdriver:stackdriver_utils_test
```

clang is 4.0.1-10

**Comment by isturdy** [comment]
Thanks for the catch, and the tip on using clang-3.8.

**Pull request merged**